core/src/main/java/org/apache/accumulo/core/clientImpl/bulk/LoadMappingIterator.java

```diff
@@ -72,7 +74,23 @@ public boolean hasNext() {
 
   @Override
   public Map.Entry<KeyExtent,Bulk.Files> next() {
+    if (!hasNext()) {
+      throw new NoSuchElementException();
+    }
```

Comment:
I think this is generally the user's responsibility to call. It doesn't actually protect much, because next() is not thread safe, and another thread could call next() successfully, and grab the last element, while this thread successfully makes it past this check and then immediately fails, just like it would have before this change.

A better fix would be to surround the gson.fromJson with a try-catch and handle the exception it throws when there's no more elements in the reader, and convert that exception to a NoSuchElementException in the catch block for that.

Reply:
This comment was incorrectly marked resolved without being addressed.